### PR TITLE
Feature/mds 3610 allow users to change between now types

### PR DESCRIPTION
**Pull request by mhuseinov**
- edit Notice of work type on technical review for NOW,
- edit Notice of work type on Draft permit,
- generate NTR document if NOW type changed,
- edit NOW type on ADA,
- validation of Permit and NOW type mismatch.

note: test cases are in jira ticket comments 

Screenshot 2021-05-11 131229
Screenshot 2021-05-11 131343
Screenshot 2021-05-11 131504
Screenshot 2021-05-11 131919
Screenshot 2021-05-11 132721
Screenshot 2021-05-11 132827

### CODE REVIEW

File: services/core-web/src/HOC/ApplicationGuard.js [hunk 1]
```diff
@@ -101,10 +103,14 @@ export const ApplicationGuard = (WrappedComponent) => {
       await Promise.all([
         this.props.fetchImportedNoticeOfWorkApplication(id).then(({ data }) => {
           this.setState({ mineGuid: data.mine_guid });
+          this.props.fetchDraftPermitByNOW(data.mine_guid, id);
           if (data.application_type_code === "NOW") {
             this.props.fetchOriginalNoticeOfWorkApplication(id);
             this.props.fetchImportNoticeOfWorkSubmissionDocumentsJob(id);
           }
+          if (data.application_type_code === "ADA") {
+            this.props.fetchPermits(data.mine_guid);
```
Comment (mmumby) [inline services/core-web/src/HOC/ApplicationGuard.js:112]: Don't both ADA and NOW need the permits?

File: services/core-web/src/components/noticeOfWork/applications/review/ReviewNOWApplication.js [hunk 2]
```diff
@@ -82,208 +85,223 @@ export const ReviewNOWApplication = (props) => {
     return codeHash[value];
   };
 
-  const renderApplicationInfo = () => (
```
Comment (mmumby) [inline services/core-web/src/components/noticeOfWork/applications/review/ReviewNOWApplication.js:85]: Is this file mostly formatting changes?

File: services/core-web/src/components/noticeOfWork/NOWProgressActions.js [hunk 3]
```diff
@@ -136,8 +140,8 @@ export class NOWProgressActions extends Component {
       content: modalConfig.NOW_PROGRESS_MODAL,
     });
 
-  openDraftPermitProgressModal = () =>
-    this.props.openModal({
+  openDraftPermitProgressModal = () => {
+    return this.props.openModal({
```
Comment (mmumby) [inline services/core-web/src/components/noticeOfWork/NOWProgressActions.js:144]: Can you revert this back to an implicit return?

**Review by mmumby — COMMENTED** [review]
Nice work! 🔥  Just a few small questions

File: services/core-web/src/HOC/ApplicationGuard.js (hunk 1 above)
Comment (mhuseinov) [reply services/core-web/src/HOC/ApplicationGuard.js:112]: nope, they both require the draft permits and it is done in line 106

File: services/core-web/src/components/noticeOfWork/applications/review/ReviewNOWApplication.js (hunk 2 above)
Comment (mhuseinov) [reply services/core-web/src/components/noticeOfWork/applications/review/ReviewNOWApplication.js:85]: yep, except line 89-93 and passing some parameters to components

File: services/core-web/src/components/noticeOfWork/NOWProgressActions.js (hunk 3 above)
Comment (mhuseinov) [reply services/core-web/src/components/noticeOfWork/NOWProgressActions.js:144]: done

File: services/core-web/src/HOC/ApplicationGuard.js (hunk 1 above)
Comment (mmumby) [reply services/core-web/src/HOC/ApplicationGuard.js:112]: Is this for the dropdown list of permits on the draft permit modal?

Comment (mmumby) [reply services/core-web/src/HOC/ApplicationGuard.js:112]: if so, NoW amendments need that data as well.

Comment (mhuseinov) [reply services/core-web/src/HOC/ApplicationGuard.js:112]: Nope, this is not for drop-down, this is for check if the type field should be available for edit. The draft permit modal fetches this data itself on load.

**Review by mmumby — APPROVED** [review]

**Review by mmumby — APPROVED** [review]

**Pull request merged**